eglot-jl.el

```diff
@@ -90,5 +116,20 @@ Otherwise returns nil"
                ;; function instead of strings to find project dir at runtime
                '(julia-mode . eglot-jl--ls-invocation)))
 
+;;;###autoload
+(defun eglot-jl-compile-sysimage ()
+  "Create a compiled system image for the language server.
+This reduces subsequent start-up times."
+  (interactive)
+  (let ((buffer (get-buffer-create "*eglot-jl sysimage compilation*")))
```

Comment:
You can probably use the `compile` emacs function which should cut down some of this boiler-plate and make this asynchronous for free.

Reply:
One of the upsides of re-using `compile` (or the low-level `compilation-start`) would be to get correct treatment of ansi color sequences.

However, one big downside (IMO) is that `compile` & co are designed to be used interactively by humans. As such, they handle shell commands presented in the form of a string that has to be parsed. In order for us to use such an API, we'd have to first quote all arguments and join them together, which I find less safe (and reduces the gain in terms of boilerplate)

We can however easily put the sysimage compilation buffer in `view-mode` in order to benefit from standard key bindings traditionally associated to read-only buffers (e.g. <kbd>q</kbd> to bury the buffer).